src/lcec_el7041_1000.c

```diff
@@ -0,0 +1,710 @@
+  unsigned int set_count_pdo_bp;
+  unsigned int set_count_val_pdo_os;
+  unsigned int set_count_done_pdo_os;
+  unsigned int set_count_done_pdo_bp;
+  unsigned int expol_stall_pdo_os;
+  unsigned int expol_stall_pdo_bp;
+  unsigned int ina_pdo_os;
+  unsigned int ina_pdo_bp;
+  unsigned int inb_pdo_os;
+  unsigned int inb_pdo_bp;
+  unsigned int inc_pdo_os;
+  unsigned int inc_pdo_bp;
+  unsigned int inext_pdo_os;
+  unsigned int inext_pdo_bp;
+  unsigned int sync_err_pdo_os;
+  unsigned int sync_err_pdo_bp;
+  unsigned int tx_toggle_pdo_os;
+  unsigned int tx_toggle_pdo_bp;
+  unsigned int count_overflow_pdo_os;
+  unsigned int count_overflow_pdo_bp;
+  unsigned int count_underflow_pdo_os;
+  unsigned int count_underflow_pdo_bp;
+  unsigned int latch_c_valid_pdo_os;
+  unsigned int latch_c_valid_pdo_bp;
+  unsigned int latch_ext_valid_pdo_os;
+  unsigned int latch_ext_valid_pdo_bp;
+  unsigned int ena_latch_c_pdo_os;
+  unsigned int ena_latch_c_pdo_bp;
+  unsigned int ena_latch_ext_pos_pdo_os;
+  unsigned int ena_latch_ext_pos_pdo_bp;
+  unsigned int ena_latch_ext_neg_pdo_os;
+  unsigned int ena_latch_ext_neg_pdo_bp;
+  unsigned int count_pdo_os;
+  unsigned int latch_pdo_os;
+
+  unsigned int dcm_ena_pdo_os;
+  unsigned int dcm_ena_pdo_bp;
+  unsigned int dcm_reset_pdo_os;
+  unsigned int dcm_reset_pdo_bp;
+  unsigned int dcm_reduce_torque_pdo_os;
+  unsigned int dcm_reduce_torque_pdo_bp;
+  unsigned int dcm_velo_pdo_os;
+  unsigned int dcm_ready_to_enable_pdo_os;
+  unsigned int dcm_ready_to_enable_pdo_bp;
+  unsigned int dcm_ready_pdo_os;
+  unsigned int dcm_ready_pdo_bp;
+  unsigned int dcm_warning_pdo_os;
+  unsigned int dcm_warning_pdo_bp;
+  unsigned int dcm_error_pdo_os;
+  unsigned int dcm_error_pdo_bp;
+  unsigned int dcm_move_pos_pdo_os;
+  unsigned int dcm_move_pos_pdo_bp;
+  unsigned int dcm_move_neg_pdo_os;
+  unsigned int dcm_move_neg_pdo_bp;
+  unsigned int dcm_torque_reduced_pdo_os;
+  unsigned int dcm_torque_reduced_pdo_bp;
+  unsigned int dcm_din1_pdo_os;
+  unsigned int dcm_din1_pdo_bp;
+  unsigned int dcm_din2_pdo_os;
+  unsigned int dcm_din2_pdo_bp;
+  unsigned int dcm_sync_err_pdo_os;
+  unsigned int dcm_sync_err_pdo_bp;
+  unsigned int dcm_tx_toggle_pdo_os;
+  unsigned int dcm_tx_toggle_pdo_bp;
+
+  int enc_do_init;
+  int16_t enc_last_count;
+  double enc_old_scale;
+  double enc_scale_recip;
+  double dcm_old_scale;
+  double dcm_scale_recip;
+
+} lcec_el7041_1000_chan_t;
+
+typedef struct {
+  lcec_el7041_1000_chan_t chans[LCEC_EL7041_1000_CHANS];
+  int last_operational;
+
+  hal_bit_t last_dcm_enable;
+
+  hal_bit_t *fault;
+  hal_bit_t *fault_reset;
+  hal_u32_t fault_autoreset_cycles;
+  hal_u32_t fault_autoreset_retries;
+
+  hal_bit_t internal_fault;
+
+  hal_u32_t fault_reset_retry;
+  hal_u32_t fault_reset_state;
+  hal_u32_t fault_reset_cycle;
+
+} lcec_el7041_1000_data_t;
+
+// Auto generated with "$ ethercat -p [EL7041-1000 index] cstruct"
+
+static ec_pdo_entry_info_t lcec_el7041_1000_pdo_entries[] = {
```

Comment:
I think it would be better to use separate structures instead of reference members by offsets.